Printing integers

bmd2001 opened the pull request.

bmd2001 committed:
Print integers for macos x86_64

bmd2001 committed:
Added Subroutines

bmd2001 committed:
Updated Unit Tests for print subroutines

Anon0x19 commented on src/compiler/generator/subroutines/int_to_str.rs:
```diff
@@ -0,0 +1,60 @@
+pub fn get_int_to_str_subroutine() -> String{
+    match TARGET_ARCH {
+        Arch::X86_64 => {get_int_to_str_x86_64()}
+        Arch::AArch64 => {get_int_to_str_aarch64()}
```
.

Anon0x19 commented on src/compiler/generator/subroutines/print.rs:
```diff
@@ -0,0 +1,119 @@
+        Arch::AArch64 => {
+            match TARGET_OS {
+                OS::Linux => get_print_aarch64_linux(),
+                _ => get_print_aarch64_mac(),
```
.

Anon0x19 commented on src/compiler/generator/subroutines/print.rs:
```diff
@@ -0,0 +1,119 @@
+        Arch::X86_64 => {
+            match TARGET_OS {
+                OS::Linux => get_print_x86_64_linux(),
+                _ => get_print_x86_64_mac(),
```
.

Anon0x19 reviewed (changes requested):
Looks great so far! It might be worth waiting until we can get the windows compatibility [pr](https://github.com/bmd2001/ProgrammingLanguage/pull/48) merged and having a look to see if the (dots) need any changes. However, if you prefer, we can get this merged ASAP.

bmd2001 committed:
Fixed rebasing issues

bmd2001 committed:
Modified Integration test to check for printing

bmd2001 committed:
Modified load from label instruction

Changed pushing and popping on macOS arm

Changed offset logic

Changed exit

Fixed wrong register usage

Added newline to arm mac

Definitive 0 issue solution when printing numbers

bmd2001 committed:
Added alignment for arm on Linux

bmd2001 committed:
Changed printing for arm Linux

bmd2001 committed:
Test

bmd2001 committed:
x86_64 printing bug fix

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
Changed subroutines for windows

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test

bmd2001 committed:
test
